js/Update.elm

```diff
@@ -655,8 +655,12 @@ tryUpdate :
     -> model
     -> Eff model
 tryUpdate lens upd model =
-    Lens.update lens upd model
-        |> Maybe.withDefault (Debug.log "Cannot update" model ! [])
+    case Lens.update lens upd model of
```

Comment:
The reason it was throwing errors was that Elm uses eager-evaluation, which means that arguments to functions are evaluated as much as possible, until they reach a case expressions or a lambda, before they are passed into the functions. So, `Debug.log "Cannot update" model` was being evaluated every time even though we never executed the `Nothing` branch inside `Maybe.withDefault`. To prevent this the only way is to explicit write out the case expression.

Reply:
Gah, that's annoying. Good catch though.